dbms/src/Storages/MergeTree/MergeTreeReadPool.cpp

```diff
@@ -22,8 +26,11 @@ MergeTreeReadPool::MergeTreeReadPool(
     const bool do_not_steal_tasks)
     : backoff_settings{backoff_settings}, backoff_state{threads}, data{data},
       column_names{column_names}, do_not_steal_tasks{do_not_steal_tasks},
-      predict_block_size_bytes{preferred_block_size_bytes > 0}, prewhere_info{prewhere_info}
+      predict_block_size_bytes{preferred_block_size_bytes > 0}, prewhere_info{prewhere_info}, parts_ranges{parts}
 {
+    /// reverse from right-to-left to left-to-right
+    for (auto & part_ranges : parts_ranges)
```

Comment:
Who says that part_ranges was in reversed order?
Maybe better to reverse them outside of MergeTreeReadPool constructor?

Reply:
Yes comment is bad, I'll fix it. But I don't see any problems with reverse in constuctor.  MergeTreeSelectExecutor creates single MergeTreeReadPool per request. What wrong with that reverse?

Reply:
It's non obvious, that we expect reversed array in constructor.

Reply:
> because 'reverse' was done in MergeTreeDataSelectExecutor

It's non obvious that `MergeTreeDataSelectExecutor` is the single place where `MergeTreeReadPool` is created.